src/dataframe/insertion.jl

```diff
@@ -762,16 +766,32 @@ function _dfr_row_inserter!(df::DataFrame, loc::Integer, dfr::DataFrameRow,
     return _row_inserter!(df, loc, dfr, mode, cols, promote, nrows)
 end
 
+# Correctness of this conversion is ensured because dictionary constructors
+# do not perform promotion of values types
+_normalize_abstractrow(row::Tables.AbstractRow) =
+    Tables.OrderedCollections.OrderedDict(n => Tables.getcolumn(row, n) for
+                                          n in Tables.columnnames(row))
```

Comment:
Doesn't creating an `OrderedDict` incur a large performance penalty?

Also, I think we should take a direct dependency on OrderedCollections, as there's no guarantee that Tables.jl continues to import it in the future.

Reply:
Good point, as usual 😄. I have removed the need for any conversion in the implementation (so also there is no need for dependency).

@ararslan - can you please re-check if all works with Legolas.jl? Thank you!

Reply:
> can you please re-check if all works with Legolas.jl? Thank you!

Works as expected, thanks!